bin/git-files.sh

```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+git ls-tree -r --name-only HEAD -z | TZ=UTC xargs -0n1 -I_ git --no-pager log -1 --date=iso-local --format="%ad _" -- _
```

Comment:
@Shopify/cli-foundations this one is a bit slow and blocks the execution of `lint` for seconds. Any idea of how to make this one more performant?

Reply:
what do you think if we instead checked the last modified time of the offending file when the error arises? right now we are paying the price of reading the last modified times of EVERY file while we only care about a small subset of them

Reply:
Let me do some benchmarking. Getting the last modified date for each file will require a spawning a process which is also expensive to do for every file. Perhaps less than doing all the files together.

Reply:
It's orders of magnitude faster. Let me know what you think @Arkham